config/kernel-acl.m4

```diff
@@ -263,11 +275,18 @@ AC_DEFUN([ZFS_AC_KERNEL_INODE_OPERATIONS_SET_ACL], [
 		AC_DEFINE(HAVE_SET_ACL, 1, [iops->set_acl() exists])
 		AC_DEFINE(HAVE_SET_ACL_USERNS, 1, [iops->set_acl() takes 4 args])
 	],[
-		ZFS_LINUX_TEST_RESULT([inode_operations_set_acl], [
+		ZFS_LINUX_TEST_RESULT([inode_operations_set_acl_userns_dentry], [
 			AC_MSG_RESULT(yes)
-			AC_DEFINE(HAVE_SET_ACL, 1, [iops->set_acl() exists, takes 3 args])
+			AC_DEFINE(HAVE_SET_ACL, 1, [iops->set_acl() exists])
+			AC_DEFINE(HAVE_SET_ACL_USERNS_DENTRY_ARG2, 1,
+			    [iops->set_acl() takes 4 args, arg2 is struct dentry *])
 		],[
-			AC_MSG_RESULT(no)
+			ZFS_LINUX_TEST_RESULT([inode_operations_set_acl], [
+				AC_MSG_RESULT(yes)
+				AC_DEFINE(HAVE_SET_ACL, 1, [iops->set_acl() exists, takes 3 args])
+			],[
+				AC_MSG_RESULT(no)
```

Comment:
We should switch to `ZFS_LINUX_REQUIRE_API([i_op->set_acl()], [3.14])` here from `AC_MSG_RESULT(no)`. That is a new feature from 556ed0953712bd94b49363c7c596d3d8000ff874. It will break the build on new kernel versions when the API does not match, while letting the build process continue to work on older kernels that do not have this API. If had already been in place here, we would have realized that the API was broken much sooner.

Reply:
Thanks, I've updated the commit to reflect this new addition, as well